contracts/clients/src/BlsWalletWrapper.ts

```diff
@@ -65,22 +64,23 @@ export default class BlsWalletWrapper {
    * @param privateKey private key associated with the wallet
    * @param verificationGatewayAddress address of the VerficationGateway contract
    * @param signerOrProvider ethers.js Signer or Provider
-   * @param blsWalletSigner (optional) a BLS Wallet signer
    * @returns The wallet's address
    */
   static async Address(
     privateKey: string,
     verificationGatewayAddress: string,
     signerOrProvider: SignerOrProvider,
-    blsWalletSigner?: BlsWalletSigner,
```

Comment:
Previously, the `privateKey` argument was used to derive the `pubKeyHash` and run `validateWalletNotRecovered`. We no longer need to pass the `blsWalletSigner` in, as we can achieve the same by just passing the `privateKey` and `signerOrProvider` into a new instance of a `BlsWalletSigner`.

Another reason to remove it would be, to avoid scenarios where the `blsWalletSigner` passed into this method is bound to a different `privateKey` than the dedicated `privateKey` argument. Which, may be confusing as you'd essentially be passing in 2 private keys to the method, and the address would be dervied from the `blsWalletSigner`'s private key, not the dedicated `privateKey` arg.

___

I was hesitant about doing this first, as it seemed a more significant change to the API, but will raise for discussion:

Option 2 would be to keep in line with the data abstraction theme of this PR and remove the `privateKey` arg and do something like this:

```ts
static async Address(
    blsWalletSigner: BlsWalletSigner,
    verificationGatewayAddress: string,
    signerOrProvider: SignerOrProvider,
  ): Promise<string> {
    const verificationGateway = VerificationGateway__factory.connect(
      verificationGatewayAddress,
      signerOrProvider,
    );
    const pubKeyHash = blsWalletSigner.getPublicKeyHash();

    const existingAddress = await verificationGateway.walletFromHash(
      pubKeyHash,
    );
    const hasExistingAddress = !BigNumber.from(existingAddress).isZero();
    if (hasExistingAddress) {
      return existingAddress;
    }

    const expectedAddress = await this.ExpectedAddress(
      verificationGateway,
      pubKeyHash,
    );
    this.validateWalletNotRecovered(
      blsWalletSigner,
      verificationGateway,
      expectedAddress,
    );

    return expectedAddress;
}
```